test/test_framework.py

```diff
@@ -1084,12 +1113,19 @@ def test_mutable_types(self):
         # Test and validation functions in a list of 2-tuples.
         # Assignment and keywords like del are not supported in lambdas
         #  so functions are used instead.
-        test_operations = [(
-            lambda: {},         # Operand A.
-            None,               # Operand B.
-            {},                 # Expected result.
-            lambda a, b: None,  # Operation to perform.
-            lambda res, expected_res: self.assertEqual(res, expected_res)  # Validation to perform.
+        test_case = typing.Tuple[
+            typing.Callable[[], typing.Any],                        # Called to get operand A.
+            typing.Any,                                             # Operand B.
+            typing.Any,                                             # Expected result.
+            typing.Callable[[typing.Any, typing.Any], None],        # Operation to perform.
+            typing.Callable[[typing.Any, typing.Any], typing.Any],  # Validation to perform.
```

Comment:
This is definitely one of those cases where IMO the 5-tuple of lambdas isn't helping clarity, and we should probably just spell these out as separate tests. The use of tuples with multiple statements and the need to use 
tricks like `a.__setitem__(1, b)` instead of `a[1] = b` is a bit of a giveaway. Probably just leave it as is for now though to avoid churn.

Thanks for adding the comment, that helps. :-)

Let's tweak the comment to just say "list of tuples" instead of "list of 2-tuples" though while we're here though. 2-tuples seems old/wrong.

Reply:
> This is definitely one of those cases where IMO the 5-tuple of lambdas isn't helping clarity, and we should probably just spell these out as separate tests.

Agreed! I almost did this when I came across it, after we'd done something similar in one of the other test modules, and it just seemed like maybe it was a bit too much code to re-write in what would essentially be a drive-by.

> The use of tuples with multiple statements and the need to use tricks like `a.__setitem__(1, b)` instead of `a[1] = b` is a bit of a giveaway.

Yeah, and that "operand A" has to be a lambda to get the value but "operand B" is just the value, and even that all the lambdas don't return anything (other than None) and are pure side-effects.

> Probably just leave it as is for now though to avoid churn.

:+1: